operators/multiclusterobservability/api/shared/multiclusterobservability_shared.go

```diff
@@ -49,6 +49,14 @@ type ObservabilityAddonSpec struct {
 	// +kubebuilder:default:=1073741824
 	ScrapeSizeLimitBytes int `json:"scrapeSizeLimitBytes,omitempty"`
 
+	// Workers is the number of workers that work parallelly to push metrics to hub server.
+	// If set to > 1, metrics-collector will shard /federate calls to Prometheus, based on
+	// matcher rules provided by allowlist.
+	// +optional
+	// +kubebuilder:default:=1
+	// +kubebuilder:validation:Minimum=1
+	Workers int32 `json:"workers,omitempty"`
```

Comment:
nit: Is it clear on the addon spec that workers corresponds to prometheus federate workers and not something else?

Reply:
I'll add a better description